steel/README.md

```diff
@@ -114,19 +114,23 @@ let journal = Journal {
 env::commit_slice(&journal.abi_encode());
 ```
 
-We also provide an example, [erc20-counter], showcasing such integration.
+We provide several examples showcasing such integration, including [erc20-counter] and [token-stats].
 
-### Block hash validation
+### Block commitment validation
 
-Since internally the `blockhash` opcode is used for validation, the commitment must not be older than 256 blocks.
-Given a block time of 12 seconds, this allows just over 50 minutes to create the proof and ensure that the validating transaction is included in a block.
-In many cases, this will work just fine: even very large computations such as proving an entire Ethereum block can be done in well under 50 minutes with sufficient resources.
+Validating the block committed by a Steel proof is essential to ensure that the proof accurately reflects the correct blockchain state.
+Steel supports two methods for block validation (see the `validateCommitment` function in [Steel.sol](../contracts/src/steel/Steel.sol)).
 
-For scenarios needing a verified block hash older than 256 blocks:
+#### 1. Block hash Commitment
+This method uses the `blockhash` opcode to commit to a block hash that is no more than 256 blocks old. With Ethereum's 12-second block time, this provides a window of about 50 minutes to generate the proof and ensure that the validating transaction is contained in a block. This approach is ideal for most scenarios, including complex computations, as it typically provides sufficient time to generate the proof.
+#### 2. Beacon Block Root Commitment
+The second method allows validation using the [EIP-4788] beacon roots contract. This technique extends the time window in which the proof can be validated on-chain to just over a day, making it suitable for scenarios requiring more extensive computation. It requires access to a beacon API endpoint and can be enabled by calling `EvmEnv::into_beacon_input`.
+However, this approach is specific to Ethereum Steel proofs and depends on the implementation of EIP-4788. Note that EIP-4788 only provides access to the parent beacon root, requiring iterative queries in Solidity to retrieve the target beacon root for validation. This iterative process can result in slightly higher gas costs compared to using the `blockhash` opcode. Overall, it is suitable for environments where longer proof generation times are required.
 
-* Save the required block hash to the contract state if known in advance (e.g., when initiating a governance proposal).
-* Use RISC Zero to prove the hash chain from the queried block up to a block within the most recent 256.
+A *bookmarking* validation technique can also be built on top of this. The idea is to store the target block hash in the contract state before generating a Steel proof that targets that specific block. Once the block hash (or beacon root) has been bookmarked, it can be used later for validation, ensuring that the proof corresponds to the correct blockchain state.
```

Comment:
With how this is written now, it might be implied that the bookmarking technique is only available for use with consensus block commitments, since it is under that heading.

Reply:
Hope you don't mind. I pushed 919d724d7946f4291f3f043bfe534c4ba52452aa to break up the long lines, and 3c2f455088b936ab086e70e311b54f86fbc11931 to add a "Bookmarking" heading and adjust working slightly to make it clear that bookmarking applied in both cases.